ui-react/src/components/accounts/EditImporter.tsx

```diff
@@ -17,18 +17,37 @@ export default function EditImporter({
 }: EditImporterProps) {
     const [loginFields, setLoginFields] = useState<Record<string, string>>(importer.loginFields || {});
     const [active, setActive] = useState<boolean>(importer.active);
+    const [validated, setValidated] = useState(false);
     const onSaveClicked = async () => {
         await handleSave({
             ...importer,
             active,
             loginFields
         });
     };
+    const checkFieldValidity = (loginFieldName: string, value) : boolean => {
+        return value.length >= LOGIN_FIELD_MIN_LENGTH[loginFieldName];
+    };
     const onLoginFieldChanged = (loginFieldName: string, value) => {
         setLoginFields({
             ...loginFields,
             [loginFieldName]: value
         });
+
+        if (!checkFieldValidity(loginFieldName, value)) {
+            setValidated(false);
+            return;
+        }
+        
+        for (let fieldKey in loginFields) {
+            if (loginFieldName === fieldKey) continue
+
+            if (!checkFieldValidity(fieldKey, loginFields[fieldKey])) {
+                setValidated(false);
+                return;
+            }
+        }
```

Comment:
This code is a little strange for me. Here is what I understand:

1. First, you're trying to validate the current field, and if it is OK, you're validating the others (by skipping the current).  
 I'm not sure what the idea is, why not validating all the fields in one loop?
2. You're right, this validation should happend every time a field is changed. But in React, we can do it with `useEffect`, and implementing this way is better, because it will depend the validation in the state change, and not in the input event.

Reply:
@baruchiro 
Sorry for interrupting, but I am not sure this is the right use case for `useEffect`. 
have you seen this video? really changed how i look at this hook
https://www.youtube.com/watch?v=HPoC-k7Rxwo

Reply:
Thanks for the video!
@baruchiro Already implemented the `useEffect` but didn't push it yet.
Let me know what you'd like after seeing the video.

Regarding your first question, I wasn't validating all the fields in one loop because the changed field didn't have the updated value (it had the previous version- e.g. you type `c` after `ab` but it only sees `ab`). Probably because `setLoginFields` didn't finish setting it and there was some race condition.

Reply:
@avivnakar Hi! Nice to see you're participating in our reviews :-)

Usually, I wouldn't say I like videos to learn something new, but I will try to watch for you two.

Reply:
OK, I sew it and I accept. I don't think it is critical here, and sometimes the readability of the code is important, but let's use **EventHandler** instead of **useEffect** here.

But @ShiranAbir, I still want you to refactor the code to just do one check on the object, for example: `setValidated(Object.entries(loginFields).some(([key, value]) => !checkFieldValidity(key, value)))`

Reply:
@baruchiro Done. Please approve and merge.